tedana/tests/test_integrity _robustica.py

```diff
@@ -0,0 +1,667 @@
+"""Integration tests for "real" data."""
+
+import glob
+import json
+import logging
+import os
+import os.path as op
+import re
+import shutil
+import subprocess
+import tarfile
+from datetime import datetime
+from gzip import GzipFile
+from io import BytesIO
+
+import pandas as pd
+import pytest
+import requests
+from pkg_resources import resource_filename
+
+from tedana.io import InputHarvester
+from tedana.workflows import t2smap as t2smap_cli
+from tedana.workflows import tedana as tedana_cli
+from tedana.workflows.ica_reclassify import ica_reclassify_workflow
+
+# Need to see if a no BOLD warning occurred
+LOGGER = logging.getLogger(__name__)
+# Added a testing logger to output whether or not testing data were downlaoded
+TestLGR = logging.getLogger("TESTING")
+
+
+def check_integration_outputs(fname, outpath, n_logs=1):
+    """
+    Checks outputs of integration tests.
+
+    Parameters
+    ----------
+    fname : str
+        Path to file with expected outputs
+    outpath : str
+        Path to output directory generated from integration tests
+    """
+
+    # Gets filepaths generated by integration test
+    found_files = [
+        os.path.relpath(f, outpath)
+        for f in glob.glob(os.path.join(outpath, "**"), recursive=True)[1:]
+    ]
+
+    # Checks for log file
+    log_regex = "^tedana_[12][0-9]{3}-[0-9]{2}-[0-9]{2}T[0-9]{2}[0-9]{2}[0-9]{2}.tsv$"
+    logfiles = [out for out in found_files if re.match(log_regex, out)]
+    assert len(logfiles) == n_logs
+
+    # Removes logfiles from list of existing files
+    for log in logfiles:
+        found_files.remove(log)
+
+    # Compares remaining files with those expected
+    with open(fname, "r") as f:
+        expected_files = f.read().splitlines()
+    expected_files = [os.path.normpath(path) for path in expected_files]
+
+    if sorted(found_files) != sorted(expected_files):
+        expected_not_found = sorted(list(set(expected_files) - set(found_files)))
+        found_not_expected = sorted(list(set(found_files) - set(expected_files)))
+
+        msg = ""
+        if expected_not_found:
+            msg += "\nExpected but not found:\n\t"
+            msg += "\n\t".join(expected_not_found)
+
+        if found_not_expected:
+            msg += "\nFound but not expected:\n\t"
+            msg += "\n\t".join(found_not_expected)
+        raise ValueError(msg)
+
+
+def data_for_testing_info(test_dataset=str):
```

Comment:
```suggestion
def data_for_testing_info(test_dataset: str):
```

If this is supposed to be a type hint, I think the syntax is incorrect. It looks to me like with your syntax, the default contents of variable "test_dataset" will be the _inbuilt function_ called "`str()`".